### PR TITLE
[14.0] webservice: add api key and public auth support

### PR DESCRIPTION
**Pull request by simahawk**
ref: cos-3712

### CODE REVIEW

File: webservice/models/webservice_backend.py [hunk 1]
```diff
@@ -14,8 +16,19 @@ class WebserviceBackend(models.Model):
...
+            ("none", "Public"),
+            ("user_pwd", "Username & password"),
+            ("api_key", "API Key"),
+        ],
+        default="user_pwd",
+        required=True,
+    )
+    username = fields.Char(auth_type="user_pwd")
+    password = fields.Char(auth_type="user_pwd")
+    api_key = fields.Char(string="API Key", auth_type="api_key")
```
Comment (TDu) [inline webservice/models/webservice_backend.py:30]: I don't understand how this `auth_type` parameter on the filed declaration works and where it comes from ?

Comment (simahawk) [reply webservice/models/webservice_backend.py:30]: I'm adding it. See `_valid_field_parameter` override

Comment (TDu) [reply webservice/models/webservice_backend.py:30]: Thanks, interesting.

**Review by TDu — APPROVED** [review]

**Review by etobella — APPROVED** [review]
Makes sense to me

**Pull request merged**